src/test/fuzz/banman.cpp

```diff
@@ -93,6 +95,22 @@ FUZZ_TARGET_INIT(banman, initialize_banman)
                     ban_man.Discourage(ConsumeNetAddr(fuzzed_data_provider));
                 });
         }
+        if (!force_read_and_write_to_err) {
+            ban_man.DumpBanlist();
+            SetMockTime(ConsumeTime(fuzzed_data_provider));
+            banmap_t banmap;
+            ban_man.GetBanned(banmap);
+            BanMan ban_man_read{banlist_file, nullptr, offset};
+            banmap_t banmap_read;
+            ban_man_read.GetBanned(banmap_read);
+            assert(banmap.size() == banmap_read.size());
+            auto it1{banmap.begin()};
+            auto it2{banmap_read.begin()};
+            for (; it1 != banmap.end(); ++it1, ++it2) {
```

Comment:
This may brick if `banmap_t` is changed from `std::map` to `std::unordered_map`. It should be changed because the latter is faster and we don't need an order into this container.

Maybe add something like `static_assert(std::is_base_of<std::map, banmap_t>::value)` to ensure this gets adjusted if `banmap_t` is changed in the future?

Or a slower, but more robust variant: for each entry in `banmap` ensure that it can be found in `banmap_read` and that both maps have the same size.

Reply:
```diff
             assert(banmap.size() == banmap_read.size());
-            auto it1{banmap.begin()};
-            auto it2{banmap_read.begin()};
-            for (; it1 != banmap.end(); ++it1, ++it2) {
-                assert(it1->first == it2->first);
-                assert(it1->second.ToJson().write() == it2->second.ToJson().write());
+            for (const auto& [subnet, ban_entry] : banmap) {
+                const auto it = banmap_read.find(subnet);
+                assert(it != banmap_read.end());
+                assert(it->second.ToJson().write() == ban_entry.ToJson().write());
             }
```

Reply:
Or even better:
```diff
diff --git i/src/test/fuzz/banman.cpp w/src/test/fuzz/banman.cpp
index 6aa95b484f..8a911f22fb 100644
--- i/src/test/fuzz/banman.cpp
+++ w/src/test/fuzz/banman.cpp
@@ -29,12 +29,17 @@ int64_t ConsumeBanTimeOffset(FuzzedDataProvider& fuzzed_data_provider) noexcept
 
 void initialize_banman()
 {
     static const auto testing_setup = MakeNoLogFileContext<>();
 }
 
+bool operator==(const CBanEntry& lhs, const CBanEntry& rhs) {
+    return lhs.nVersion == rhs.nVersion && lhs.nCreateTime == rhs.nCreateTime &&
+           lhs.nBanUntil == rhs.nBanUntil;
+}
+
 FUZZ_TARGET_INIT(banman, initialize_banman)
 {
     // The complexity is O(N^2), where N is the input size, because each call
     // might call DumpBanlist (or other methods that are at least linear
     // complexity of the input size).
     int limit_max_ops{300};
@@ -100,18 +105,12 @@ FUZZ_TARGET_INIT(banman, initialize_banman)
             SetMockTime(ConsumeTime(fuzzed_data_provider));
             banmap_t banmap;
             ban_man.GetBanned(banmap);
             BanMan ban_man_read{banlist_file, nullptr, offset};
             banmap_t banmap_read;
             ban_man_read.GetBanned(banmap_read);
-            assert(banmap.size() == banmap_read.size());
-            auto it1{banmap.begin()};
-            auto it2{banmap_read.begin()};
-            for (; it1 != banmap.end(); ++it1, ++it2) {
-                assert(it1->first == it2->first);
-                assert(it1->second.ToJson().write() == it2->second.ToJson().write());
-            }
+            assert(banmap_read == banmap);
         }
     }
     fs::remove(banlist_file.string() + ".dat");
     fs::remove(banlist_file.string() + ".json");
 }
```

Reply:
Made it a compile failure to change the type without touching this file

Reply:
This is now addressed in another way. Just out of curiosity - why not the `assert(banmap_read == banmap);` variant? It is shorter and works for both `map` and `unordered_map`.